tuareg.el

```diff
@@ -3645,35 +3682,43 @@ It is assumed that the range START-END delimit valid OCaml phrases."
 (defun tuareg-eval-region (start end)
   "Eval the current region in the OCaml REPL."
   (interactive "r")
-  (setq tuareg-interactive-last-phrase-pos-in-source start)
-  (save-excursion
-    (goto-char start)
-    (tuareg-backward-beginning-of-defun)
-    (setq start (point))
-    (setq end (cdr (tuareg-region-of-defun end)))
-    (if end
-        (tuareg-interactive--send-region start end)
-      (message "The expression after the point is not well braced."))))
+  (tuareg-interactive--send-region start end))
 
 (define-obsolete-function-alias 'tuareg-narrow-to-phrase #'narrow-to-defun
   "Apr 10, 2019")
 
 (defun tuareg-eval-phrase ()
-  "Eval the surrounding OCaml phrase (or block) in the OCaml REPL."
+  "Eval the surrounding OCaml phrase (or block) in the OCaml REPL.
+If the region is active, evaluate all phrases intersecting the region."
   (interactive)
-  (let ((opoint (point)))
-    ;; Move before the comment, if we are in one.
-    (let ((ppss (syntax-ppss)))
-      (if (nth 4 ppss) (goto-char (- (nth 8 ppss) 1))))
-    (let ((phrase (tuareg-discover-phrase)))
-      (if phrase
-          (progn
-            (tuareg-interactive--send-region (car phrase) (cadr phrase))
-            (if tuareg-skip-after-eval-phrase
-              (progn (goto-char (car (cddr phrase)))
-                     (tuareg-skip-blank-and-comments))
-              (goto-char opoint)))
-        (message "The expression after the point is not well braced.")))))
+  (let ((opoint (point))
+        start end)
+    (cond
+     ((region-active-p)
+      (setq start (min (point) (mark)))
+      (setq end (max (point) (mark)))
+      ;; Extend the region at the endpoints if they are in a phrase.
+      (dolist (pos (list start end))
+        (save-excursion
+          (goto-char pos)
+          (let ((phrase (tuareg-discover-phrase)))
+            (when (and phrase (> (cadr phrase) pos))
```

Comment:
If `phrase` is nil then that's just means there isn't anything interesting at that particular region endpoint and the region doesn't need to be extended in that direction. All as it should be.

The second test is important: we don't want to capture a phrase entirely outside the user-indicated region. Example:
```
PHRASE1
               [start]
PHRASE2
               [end]
```
If the user selects `[start]`..`[end]` then that's a clear indication that `PHRASE2` should be evaluated but not `PHRASE1`; however, `tuareg-discover-phrase` at `[start]` may capture `PHRASE1` because that's how it's traditionally defined to work.

But your question was astute because it revealed a potential bug: the second condition is indeed only applicable to the first iteration of the loop; the second iteration should have a mirrored condition. I'm fixing that and clarifying the code. Thank you!

And a subtle bug I found during testing: since we move point after evaluating, we should deactivate the region first because the user doesn't expect it to be extended.